lib/kubeclient/common.rb

```diff
@@ -253,16 +254,18 @@ def watch_entities(resource_name, options = {})
       uri = @api_endpoint.merge("#{@api_endpoint.path}/#{@api_version}/#{path}")
 
       params = {}
+      format = options[:as] == :raw ? :text : :json
       WATCH_ARGUMENTS.each { |k, v| params[k] = options[v] if options[v] }
       uri.query = URI.encode_www_form(params) if params.any?
 
-      Kubeclient::Common::WatchStream.new(uri, http_options(uri))
+      Kubeclient::Common::WatchStream.new(uri, http_options(uri), format: format)
```

Comment:
perhaps this should be named `as:` too?

Reply:
Can we please do the watch changes in another PR @NickLaMuro?
That way we could get this one in faster.

Reply:
@moolitayer No problem for me.  Sorry I jumped the gun on that one, but it definitely can wait.  Will remove shortly.